frontend/src/pages/forms/Form.jsx

```diff
@@ -28,13 +30,36 @@ const Form = ({ walletId, setWalletId }) => {
   const [alertMessage, setAlertMessage] = useState('');
   const [successful, setSuccessful] = useState(false);
   useLockBodyScroll(successful);
+  const [showPopup, setShowPopup] = useState(false);
+  const [hasOpenPosition, setHasOpenPosition] = useState(false);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const checkOpenPosition = async () => {
+      if (!walletId) return;
+      
+      try {
+        const response = await fetch(`/api/has-user-opened-position?wallet_id=${walletId}`);
+        if (!response.ok) {
+          throw new Error('Failed to check position status');
+        }
+        const data = await response.json();
+        setHasOpenPosition(data.has_opened_position);
+      } catch (error) {
+        console.error('Failed to check open position:', error);
+        setError('Failed to check position status');
+      }
+    };
+
+    checkOpenPosition();
+  }, [walletId]);
```

Comment:
api requests should be done using tanstack query library (that is already part of a project), also using axios, not fetch